designs/Transitive-Dependencies.md

```diff
@@ -0,0 +1,150 @@
+* Issue: [5887](https://github.com/NuGet/Home/issues/5887)
+
+## Description: What is it?
+
+When a developer uses NuGet in a CLI or IDE environment, they want to be able to see their project’s transitive dependencies and understand how certain dependencies are included in their package graph for PackageReference projects.
+
+## Problem: What problem is this solving?
+
+Developers are often confused with top-level & transitive dependencies in their project. A developer should be able to see all the transitive packages for their project or solution & understand where they originated from. A user should be able to promote a transitive dependency quickly & easily to a top-level dependency in the case there is a security concern on the package version depended on or if promoting a top-level package will resolve a conflict. 
+
+## Why: How do we know this is a real problem and worth solving?
+
+Knowing how a package is being included in a project helps a developer diagnose many problems. It also provides them with an opportunity to understand complex dependency chains they may have within their projects. 
+There is not a great solution that exists to understand the nature of top-level packages and their transitive dependencies.
+The solution explorer in Visual Studio does the best job at allowing a user to dive into each NuGet package & see all transitive dependencies from the top-level dependency.
+
+![](../resources/TransitiveDependencies/SolutionView.png)
+
+The dotnet CLI does not provide the insight into “why” a transitive dependency is listed, although it does list everything that has been resolved.
+
+![](../resources/TransitiveDependencies/DotNetCLI.png)
+
+The Visual Studio Package Manager UI does not support the concept of showing transitive packages in any fashion.
+
+![](../resources/TransitiveDependencies/VSPMUI.png)
+
+## Success: How do we know if we have solved this problem?
+
+By measuring the number of detractors regarding transitive dependencies & promoters of this new experience in future NuGet client surveys.
+
+We will add telemetry to categorize when a top-level package is installed, and whether it was done in the context of promoting a transitive package to a top-level one.
+
+## Audience: Who are we building this for?
+
+We are building this for .NET users, both old and new, to be successful with understanding the types of NuGet dependencies in their project(s) and solution(s).
+
+## What: What does this look like in the product?
+
+![](../resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+## Context and Scope
+
+-	https://github.com/NuGet/Home/issues/5887 
+-	https://developercommunity.visualstudio.com/content/idea/538119/add-dependency-tree-view-for-packagereferences.html 
+
+## Minimal Requirements
+
+-	Visual list of transitive dependencies.
+-	Ability to see a transitive dependency package details.
+-	Ability to determine where a transitive dependency originated.
```

Comment:
Sounds like we are also trying to solve https://github.com/NuGet/Home/issues/6339 with this. 

Visual Studio 2015 had a Package Visualizer you can look into.

Reply:
I knew there was something... I think this is a separate thing entirely and can remain in that issue 💯